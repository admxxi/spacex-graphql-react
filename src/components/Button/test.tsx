import React from 'react'
import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { Button, ButtonType } from './index'

it('renders correctly', () => {
  const fn = jest.fn()
  const { container } = render(
    <Button
      type={ButtonType.primary}
      text="Load more"
      disabled={false}
      onclick={fn}
    />
  )
  expect(container.firstChild).toMatchSnapshot()
})

it('should execute the click function once', () => {
  const fn = jest.fn()
  render(
    <Button
      text="See more"
      icon={'youtube'}
      type={ButtonType.primary}
      disabled={false}
      onclick={fn}
    />
  )

  const button = screen.getByRole('button', { name: /See more/i })
  userEvent.click(button)
  expect(fn).toHaveBeenCalledTimes(1)
})
