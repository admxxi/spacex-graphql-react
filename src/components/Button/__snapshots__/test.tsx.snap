// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<button
  class="button is-large is-link is-light is-primary is-2"
  type="button"
>
  <span>
    Load more
  </span>
</button>
`;
